griptape/drivers/vector/base_vector_store_driver.py

```diff
@@ -54,7 +54,11 @@ def upsert_text_artifact(
         **kwargs,
     ) -> str:
         meta = {} if meta is None else meta
-        vector_id = utils.str_to_hash(artifact.value) if vector_id is None else vector_id
+
+        # Properly handle CSVs
+        artifact_value_string = artifact.value if isinstance(artifact.value, str) else str(artifact.value)
+
+        vector_id = utils.str_to_hash(artifact_value_string) if vector_id is None else vector_id
```

Comment:
use `.to_string()` here instead.
```suggestion
        vector_id = utils.str_to_hash(artifact.to_string()) if vector_id is None else vector_id
```

Reply:
Isn't it `to_text()`?

Reply:
yep my bad

Reply:
Why this change? Python's standard form is to convert to strings with str()? A dict is a built in type and this method will properly handle other types like arrays vs. using a custom method.

Reply:
`BaseArtifact.to_text()` is for handling `value`s in more specific ways. For example, `CsvRowArtifact.to_text()` does proper CSV parsing and generates more friendly string values.

Reply:
gotcha. is there a reason we didn't just override `__str__()`?

will make this fix, but seems weird to not use built-in functionaltiy.

Reply:
Ha, we did in fact:

```python
def __str__(self) -> str:
    return self.to_text()
```

I can't remember the reason we have a separate `to_text()` method. But there was a reason.